lib/main.dart

```diff
@@ -25,20 +25,41 @@ class CameraIssue extends StatefulWidget {
   _CameraIssueState createState() => _CameraIssueState();
 }
 
-class _CameraIssueState extends State<CameraIssue> with ForegroundStateObserver {
+class _CameraIssueState extends State<CameraIssue> with WidgetsBindingObserver {
   CameraController? _controller;
 
   @override
-  void didForeground() {
-    _initCamera();
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    // App state changed before we got the chance to initialize.
+    if (_controller == null || !_controller!.value.isInitialized) {
+      return;
+    }
+    if (state == AppLifecycleState.inactive) {
+      _controller?.dispose();
+    } else if (state == AppLifecycleState.resumed) {
+      if (_controller != null) {
+        _initCamera();
+      }
+    }
   }
 
   @override
-  void didBackground() {
-    _controller?.dispose();
-    _controller = null;
+  void initState() {
+    super.initState();
+    _initCamera();
   }
```

Comment:
Please add `WidgetsBinding.instance!.addObserver(this);` (as well as removal and controller clean upon dispose)